contracts/core/Dispatcher.sol

```diff
@@ -94,9 +90,23 @@ contract Dispatcher is OwnableUpgradeable, UUPSUpgradeable, IDispatcher {
         L1Header calldata l1header,
         OpL2StateProof calldata proof,
         uint256 height,
-        uint256 appHash
+        uint256 appHash,
+        uint256 clientId
     ) external returns (uint256 fraudProofEndTime, bool ended) {
-        return _lightClient.addOpConsensusState(l1header, proof, height, appHash);
+        LightClient lightClient = _clientIdToLightClient[clientId];
+        if (address(lightClient) == address(0)) {
+            revert IBCErrors.lightClientNotFound(clientId);
+        }
+        return lightClient.addOpConsensusState(l1header, proof, height, appHash);
+    }
+
+    function setNewConnection(string calldata connection, LightClient lightClient) external onlyOwner {
+        _setNewConnection(connection, lightClient);
+    }
+
+    function removeConnection(string calldata connection) external onlyOwner {
+        uint256 clientId = _connectionToClientId[connection];
```

Comment:
delete _connectionToClientId[connection]; ?

Reply:
should be removed now in the commit named similarly to this one